pkg/extensions/tbtc/tbtc.go

```diff
@@ -36,31 +39,65 @@ func Initialize(ctx context.Context, handle Handle) error {
 		)
 	}
 
+	err = tbtc.monitorProvideRedemptionSignature(
+		ctx,
+		exponentialBackoff,
+		105*time.Minute, // 15 minutes before the 2 hours on-chain timeout
+	)
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize provide redemption "+
+				"signature monitoring: [%v]",
+			err,
+		)
+	}
+
+	err = tbtc.monitorProvideRedemptionProof(
+		ctx,
+		exponentialBackoff,
+		345*time.Minute, // 15 minutes before the 6 hours on-chain timeout
+	)
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize provide redemption "+
+				"proof monitoring: [%v]",
+			err,
+		)
+	}
+
 	logger.Infof("tbtc extension has been initialized")
 
 	return nil
 }
 
-type depositEventHandler func(deposit string)
+type depositEventHandler func(depositAddress string)
 
 type watchDepositEventFn func(
 	handler depositEventHandler,
 ) (subscription.EventSubscription, error)
 
-type watchKeepClosedFn func(deposit string) (
+type watchKeepClosedFn func(depositAddress string) (
 	keepClosedChan chan struct{},
 	unsubscribe func(),
 	err error,
 )
 
-type submitDepositTxFn func(deposit string) error
+type submitDepositTxFn func(depositAddress string) error
 
 type backoffFn func(iteration int) time.Duration
 
+type timeoutFn func(depositAddress string) (time.Duration, error)
```

Comment:
Maybe `timeoutShiftFn`? This function is shifting the `timeout` parameter if needed.

Reply:
It's shifting only for one case. For the rest, it returns a constant timeout. I look a this as a function that returns the timeout value so I think the `timeoutFn` covers more possible use cases.